boto3/s3/inject.py

```diff
@@ -356,3 +367,267 @@ def object_copy(self, CopySource, ExtraArgs=None, Callback=None,
         CopySource=CopySource, Bucket=self.bucket_name, Key=self.key,
         ExtraArgs=ExtraArgs, Callback=Callback, SourceClient=SourceClient,
         Config=Config)
+
+
+def upload_fileobj(self, Fileobj, Bucket, Key, ExtraArgs=None,
+                   Callback=None, Config=None):
+    """Upload a file-like object to S3.
+
+    This is a managed transfer which will perform a multipart upload in
+    multiple threads if necessary.
```

Comment:
So for now I say we make a note that the fileobj should be in a bytes mode in the docstring and make this really explicit so people do not try to pass fileobj that produce strings and unicode.